plugins/BulkTracking/Tracker/Handler.php

```diff
@@ -40,18 +44,22 @@ public function onAllRequestsTracked(Tracker $tracker, RequestSet $requestSet)
 
     public function process(Tracker $tracker, RequestSet $requestSet)
     {
-        $invalidRequests = 0;
-        foreach ($requestSet->getRequests() as $request) {
+        $isAuthenticated = $this->isBulkTrackingRequestAuthenticated($requestSet);
+
+        /** @var Response $response */
+        $response = $this->getResponse();
+        $response->setIsAuthenticated($isAuthenticated);
```

Comment:
It is not done in `Response` because I am treating it as value object (or as a data structure). I could have simply passed the RequestSet or token auth to a method in `Response`, but I am of the opinion that logic such as authentication should not be handled by such objects now that DI is more or less in place. "Service classes" (in the domain driven design context) should be responsible for such logic. Especially since ideally, we don't want to access `StaticContainer` anywhere (if Handler classes are eventually stored in DI, we won't have to here).